### ISSUE TITLE
Hidden dependency issue

### ISSUE THREAD

**Issue by ebgregory** [hpcugent/easybuild-framework#1134]
(v 1.16.1)
When I use eb --force to install software with hidden dependencies easybuild does not find the hidden dependencies and installs them as non-hidden modules.

As an example, I have an easyconfig file util-wrapper-1.0.eb (appended at the bottom) that only contains hidden dependencies (zlib, ncurses, binutils) and installs nothing, using the Toolchain easyblock.

* If I first install zlib, ncurses and binutils separately by hand using eb --hidden then run eb util-wrapper-1.0.eb, it installs util-wrapper-1.0.eb as desired finding the hidden dependencies.

* If I do not first install the dependencies, Easybuild correctly installs them as hidden when I run eb util-wrapper-1.0.eb.

* However if I then re-run the install with eb --force util-wrapper-1.0.eb, then Easybuild re-installs the dependencies as non-hidden modules so in the module directory for the dependencies I have both hidden and non-hidden modules pointing to the same install directory.

Example easyconfig file util-wrapper-1.0.eb:

```
easyblock = 'Toolchain'

name = 'util-wrapper'
version = '1.0' 
homepage = 'http://nopage'

description = "A simple wrapper for utilities needed by compilers. Nothing is installed, but it     ensures the dependencies are loaded and available."

toolchain = {'name': 'dummy', 'version': 'dummy'} 

dependencies = [
             ('zlib', '1.2.8'),
             ('binutils', '2.24'),
             ('ncurses', '5.9'),
             ]       


hiddendependencies = [
             ('zlib', '1.2.8'),
             ('binutils', '2.24'),
             ('ncurses', '5.9'),
             ]       


moduleclass = 'tools'
```

**Comment by boegel** [comment]
I've seen something *very* similar w.r.t. `hiddendependencies` not being honored, but it didn't include using `--force`, so there's definitely a bug there.

Thanks for reporting @ebgregory, I'll try and get this fixed for EB v2.0.

**Comment by boegel** [comment]
fixed in #1161

**Comment by boegel** [comment]
let's follow up in this @ #1161, I want to avoid having multiple issues open for the same problem
